libs/partners/openai/langchain_openai/chat_models/base.py

```diff
@@ -335,6 +335,33 @@ def _convert_chunk_to_generation_chunk(
     return generation_chunk
 
 
+def _update_token_usage(
+    overall_token_usage: Union[int, dict], new_usage: Union[int, dict]
+):
+    # Token usage is either ints or dictionaries
+    # `reasoning_tokens` is nested inside `completion_tokens_details`
+    if isinstance(new_usage, int):
+        if not isinstance(overall_token_usage, int):
+            raise ValueError(
+                f"Got different types for token usage: "
+                f"{type(new_usage)} and {type(overall_token_usage)}"
+            )
+        return new_usage + overall_token_usage
+    elif isinstance(new_usage, dict):
+        if not isinstance(overall_token_usage, dict):
+            raise ValueError(
+                f"Got different types for token usage: "
+                f"{type(new_usage)} and {type(overall_token_usage)}"
+            )
+        return {
+            k: _update_token_usage(overall_token_usage.get(k, 0), v)
+            for k, v in new_usage.items()
```

Comment:
I would suggest inspecting the value to ensure it's an integer. The OpenAI API may add additional fields within `usage` that are not integers in the future.

Reply:
Believe this actually handles what you're asking for! Recursively calls the same function, the adding only happens if it's an int. Otherwise hits the warning "else" case at the bottom

Reply:
Agreed with @efriis , The best way to handle this by checking the instance types programmatically for `int` and `dict`. And the `else` case will then take care of warnings.

Reply:
Makes sense, thanks! I don't think you need to warn in the else condition though — we may add non-integer fields in `usage` in the future, and that's not a breaking change or unexpected.